### PR TITLE
[AN-5703] Add intent filter for Wire

### PR DESCRIPTION
**Pull request by CalumMcCall**
This enables us to auto-fill emails when new accounts are created

I left the `toOpt` as a separate commit as I wanted feedback on whether other people think it's worth adding. I'll squash it if there is agreement to keep it.
#### APK
[Download build #9668](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9668/artifact/build/artifact/wire-dev-PR1202-9668.apk)
[Download build #9672](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9672/artifact/build/artifact/wire-dev-PR1202-9672.apk)
[Download build #9808](http://192.168.10.18:8080/job/Pull%20Request%20Builder/9808/artifact/build/artifact/wire-dev-PR1202-9808.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/LaunchActivity.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+import android.content.Intent
+import com.waz.api.ClientRegistrationState.PASSWORD_MISSING
+import com.waz.api.{InitListener, Self}
+import com.waz.zclient.Intents._
+import com.waz.zclient.utils.{BackendPicker, Callback, IntentUtils}
+
+class LaunchActivity extends BaseActivity with InitListener {
+
+  val TAG = classOf[LaunchActivity].getName
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/LaunchActivity.scala]: You can drop these tags, generally only fragments need them for the fragment manager inside an activity.

File: app/src/main/scala/com/waz/zclient/LaunchActivity.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  private def persistInviteToken(): Unit = {
+    IntentUtils.getInviteToken(getIntent.getOpt).foreach { token =>
+      getControllerFactory
+        .getUserPreferencesController
+        .setGenericInvitationToken(token)
+    }
+  }
+
+  // Callbacks //////////////////////////////////////////////////
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/LaunchActivity.scala]: these comments don't help much either

File: app/src/main/scala/com/waz/zclient/MainActivity.scala [hunk 3]
```diff
@@ -273,10 +275,9 @@ class MainActivity extends BaseActivity
   private def enterApplication(self: Self): Unit = {
     error("Entering application")
     // step 1 - check if app was started via password reset intent
-    if (IntentUtils.isPasswordResetIntent(getIntent)) {
+    if (IntentUtils.isPasswordResetIntent(getIntent.toOpt)) {
```
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/MainActivity.scala]: yup, so as discussed, here we could define:

```
object PasswordResetIntent {
  def unapply(i: Intent): Boolean = {
    val data = i.getData
    data != null && WIRE_SCHEME.equals(data.getScheme) && PASSWORD_RESET_SUCCESSFUL_HOST_TOKEN.equals(data.getHost)
  }
}
```

and then do a `getIntent match { ... }` block,

but up to you

**Comment by CalumMcCall** [comment]
@deanrobertcook I've changed the last commit in this PR, do you wanna take another look before I merge?

**Comment by someoneEsle** [comment]
I would assume this PR enables us to prefill email in the registration flow, correct? I just tested signing up as a team member and I was never prompted to open a link that would take me to Wire with a prefilled email.
Also, the link that @CalumMcCall showed me via email only takes me to Wire public in the play store, so I am not even sure if I can test this on .dev.

**Comment by CalumMcCall** [comment]
The link should open the App, in this case Wire Dev, if you open it on Android. The website hasn't been updated yet. I still need to speak to someone about that flow.

**Pull request merged**